pkg/bgpv1/metrics/metrics.go

```diff
@@ -91,7 +91,8 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 	// revisit this timeout when the metrics collection starts to involve a
 	// network communication.
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-	peers, err := c.in.RouterManager.GetPeers(ctx)
+	// TODO: We need to pick the right version of BGP configuration mode. At the moment, it is hardcoded to BGPv1.
```

Comment:
Could this be figured out internally in the BGP Manager code instead of requesting bgpv1 / bgpv2 from the outside?

For example, what about relying on the `Reconcilers` / `ConfigReconcilers` in the `BGPRouterManager` to distinguish bgpv1 / bgpv2 internally?

https://github.com/cilium/cilium/blob/main/pkg/bgpv1/manager/manager.go#L91-L97

Reply:
Good point, it totally makes sense for BGP router manager to know which mode we are running in. I'll think more about the approach - will push the change today.

Reply:
Updated to move ConfigMode into its own package which can be pulled in as cell input. So later if any reconciler wants to know agent runtime state, it can use the getter provided by ConfigMode.